### ISSUE TITLE
'FastTextTrainables' object has no attribute 'vectors'

### ISSUE THREAD

**Issue by rpedela** [RaRe-Technologies/gensim#2062]
Using the [pre-trained FastText English Wikipedia model](https://github.com/facebookresearch/fastText/blob/master/pretrained-vectors.md), I load it and then save it to gensim's format. When I load the new file, I get a ```'FastTextTrainables' object has no attribute 'vectors'``` error. I am using version 3.4.0.

```python
from gensim.models import FastText
model = FastText.load_fasttext_format('./wiki.en.bin')
model.save('./gensim_en_fasttext_wiki_lg')
new_model = FastText.load('./gensim_en_fasttext_wiki_lg')
```

**Saved Files**
```
gensim_en_fasttext_wiki_lg
gensim_en_fasttext_wiki_lg.wv.vectors_ngrams.npy
gensim_en_fasttext_wiki_lg.wv.vectors.npy
```

**Error**
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/fasttext.py", line 718, in load
    model.trainables.vectors_vocab_lockf = ones(len(model.trainables.vectors), dtype=REAL)
AttributeError: 'FastTextTrainables' object has no attribute 'vectors'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "cluster_vectors.py", line 26, in <module>
    new_model = FastText.load('./gensim_en_fasttext_wiki_lg')
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/fasttext.py", line 725, in load
    return load_old_fasttext(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/deprecated/fasttext.py", line 53, in load_old_fasttext
    old_model = FastText.load(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/deprecated/word2vec.py", line 1616, in load
    model = super(Word2Vec, cls).load(*args, **kwargs)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/deprecated/old_saveload.py", line 87, in load
    obj = unpickle(fname)
  File "/usr/local/lib/python3.5/dist-packages/gensim/models/deprecated/old_saveload.py", line 380, in unpickle
    return _pickle.loads(file_bytes, encoding='latin1')
AttributeError: Can't get attribute 'FastTextVocab' on <module 'gensim.models.deprecated.fasttext' from '/usr/local/lib/python3.5/dist-packages/gensim/models/deprecated/fasttext.py'>
```

**Comment by munichma** [comment]
Same issue here, why does it save 3 different files and why do we get his error message? Pushing please!

**Comment by piskvorky** [comment]
CC @manneshiva

**Comment by munichma** [comment]
This is how my case looks like and the following error message (demonstrating from within iPython console):

`In [1]: from gensim.models.fasttext import FastText as FT_gensim`

In my directory I do have both files "wiki.en.vec" and "wiki.en.bin".

```
In [2]: %%time model = FT_gensim.load_fasttext_format("wiki.en")
CPU times: user 5min 47s, sys: 27.3 s, total: 6min 14s
Wall time: 6min 28s
```
Note: I takes extremely long to load the model in the first place. It is really annoying is that normal? What is the binary file there for, loading the gensim binary file which I generated from first reading the .vec file and than saving as Word2Vec gensim format only takes 45secs BUT it is not able to handle OOV words, therefore I really need to be able to load this model with the method showed below as I thought it should be way faster to load ones the model was build and saved.

```
In [3]: print(model)
FastText(vocab=2519370, size=300, alpha=0.025)
```

```
In [4]: %%time model.save("saved_model")
CPU times: user 8.58 s, sys: 7.73 s, total: 16.3 s
Wall time: 17.2 s
```

```
In [5]: ls | grep saved_model
saved_model
saved_model.wv.vectors.npy
saved_model.wv.vectors_ngrams.npy
```

```
In [6]: %%time loaded_model = FT_gensim.load("saved_model")
Traceback (most recent call last):

  File "<timed exec>", line 1, in <module>

  File "/anaconda2/envs/tf/lib/python3.5/site-packages/gensim/models/fasttext.py", line 725, in load
    return load_old_fasttext(*args, **kwargs)

  File "/anaconda2/envs/tf/lib/python3.5/site-packages/gensim/models/deprecated/fasttext.py", line 53, in load_old_fasttext
    old_model = FastText.load(*args, **kwargs)

  File "/anaconda2/envs/tf/lib/python3.5/site-packages/gensim/models/deprecated/word2vec.py", line 1616, in load
    model = super(Word2Vec, cls).load(*args, **kwargs)

  File "/anaconda2/envs/tf/lib/python3.5/site-packages/gensim/models/deprecated/old_saveload.py", line 87, in load
    obj = unpickle(fname)

  File "/anaconda2/envs/tf/lib/python3.5/site-packages/gensim/models/deprecated/old_saveload.py", line 380, in unpickle
    return _pickle.loads(file_bytes, encoding='latin1')

AttributeError: Can't get attribute 'FastTextVocab' on <module 'gensim.models.deprecated.fasttext' from '/anaconda2/envs/tf/lib/python3.5/site-packages/gensim/models/deprecated/fasttext.py'>

```

Side notes:
1. For what reason does the loading function want to access gensim/models/deprecated/old_saveload.py ? I do not understand this and I think this might somehow be correlated with the error message...
2.Error message comes not instantly but only after the command is already running for around 15 secs.

Thanks @piskvorky and @manneshiva for hopefully solving that issue quite soon!

**Comment by menshikh-iv** [comment]
@rpedela thanks for the report, reproduced with `python2.7` and `gensim==3.5.0`

>  I takes extremely long to load the model in the first place. It is really annoying is that normal?

That's expected, we already have an issue for it #1261 

> Why the hell does it generate 3 files?! Can you guys please explain the reason behind that? I really do not understand this!

The model contains 2 matrices (`vectors` and `vectors_ngrams`) + some model attributes `saved_model`, 3 files because we store matrices as distinct files.

**Comment by drozzy** [comment]
Any progress on this?

I hoped that saving in gensim format and loading would result in faster load times. Alas, `load` fails:

```
m = FastText.load('/src/workspace/model/cc.en.300.wv')

AttributeError: 'FastTextTrainables' object has no attribute 'vectors'
```

**Comment by menshikh-iv** [comment]
@drozzy I hope that this will be fixed soon by @mpenkov, is it?

**Comment by menshikh-iv** [comment]
Can be related to #2160

**Comment by mpenkov** [comment]
@drozzy FYI (and for my record-keeping), the bug you've reported is captured as https://github.com/RaRe-Technologies/gensim/issues/1994

**Comment by mpenkov** [comment]
@drozzy FYI (and for my record-keeping), the bug you've reported may also be related to https://github.com/RaRe-Technologies/gensim/issues/1994

**Comment by menshikh-iv** [comment]
Fixed by #2313
